bindings/pyroot/pythonizations/python/ROOT/_facade.py

```diff
@@ -9,13 +9,186 @@
+        )
+
+
+def _GetCppTypeFromNumpyType(dtype):
+    if dtype == "i4":
+        return "int"
+    elif dtype == "u4":
+        return "unsigned int"
+    elif dtype == "i8":
+        return "Long64_t"
+    elif dtype == "u8":
+        return "ULong64_t"
+    elif dtype == "f4":
+        return "float"
+    elif dtype == "f8":
+        return "double"
+
+    raise RuntimeError("Object not convertible: Python object has unknown data-type '" + dtype + "'.")
+
+
+def _AsRVec(arr):
+    """
+    Adopt memory of a Python object with array interface using an RVec.
+
+    \param[in] self self object
+    \param[in] obj PyObject with array interface
+
+    This function returns an RVec which adopts the memory of the given
+    PyObject. The RVec takes the data pointer and the size from the array
+    interface dictionary.
+    """
+    import ROOT
+
+    # Get array interface of object
+    interface = arr.__array_interface__
+
+    # Get the data-pointer
+    data = interface["data"][0]
+
+    # Get the size of the contiguous memory
+    shape = interface["shape"]
+
+    size = 0
+    for size_i in shape:
+        if size == 0:
+            size = 1
+
+        size = size * size_i
+
+    # Get the typestring and properties thereof
+    typestr = interface["typestr"]
+    if len(typestr) != 3:
+        raise RuntimeError(
+            "Object not convertible: __array_interface__['typestr'] returned '"
+            + typestr
+            + "' with invalid length unequal 3."
+        )
+    _CheckEndianessFromTypestr(typestr)
+
+    dtype = typestr[1:]
+    cppdtype = _GetCppTypeFromNumpyType(dtype)
+
+    # Construct an RVec of the correct data-type
+    klassname = "ROOT::VecOps::RVec<" + cppdtype + ">"
+
+    is_windows = os.name == "nt"
+    prefix = "0x" if is_windows else ""
+
+    address = ROOT.gInterpreter.Calc(f"new {klassname}(reinterpret_cast<{cppdtype} *>({prefix}{data}), {size})")
+
+    # Bind the object to a Python-side proxy, taking ownership of the underlying C++ object
+    out = cppyy.bind_object(address, klassname)
+    ROOT.SetOwnership(out, True)
+
+    # Bind pyobject holding adopted memory to the RVec
+    setattr(out, "__adopted__", arr)
+
+    return out
+
+
+def _MakeNumpyDataFrame(np_dict):
+    """
+    Make an RDataFrame from a dictionary of numpy arrays
+
+    \param[in] self Always null, since this is a module function.
+    \param[in] pydata Dictionary with numpy arrays
+
+    This function takes a dictionary of numpy arrays and creates an RDataFrame
+    using the keys as column names and the numpy arrays as data.
+    """
+    import ROOT
+
+    if not isinstance(np_dict, dict):
+        raise RuntimeError("Object not convertible: Python object is not a dictionary.")
+
+    if len(np_dict) == 0:
+        raise RuntimeError("Object not convertible: Dictionary is empty.")
+
+    is_windows = os.name == "nt"
+    address_prefix = "0x" if is_windows else ""
```

Comment:
```suggestion
    address_prefix = "0x" if platform.system() == "Windows" else ""
```

https://docs.python.org/3/library/platform.html#platform.system